SIDFactoryII/source/foundation/platform/sdl/platform_sdl_macos.cpp

```diff
@@ -3,196 +3,186 @@
 
 #ifndef _SF2_WINDOWS
 #ifndef _SF2_LINUX
-#include "libraries/ghc/fs_std.h"
 #include "foundation/base/assert.h"
+#include "libraries/ghc/fs_std.h"
+#include <ApplicationServices/ApplicationServices.h>
 #include <mach-o/dyld.h>
-#include <system_error>
 #include <pwd.h>
-#include <ApplicationServices/ApplicationServices.h>
+#include <system_error>
 #include <wordexp.h>
 
 using namespace fs;
 
 namespace Foundation
 {
 	PlatformSDLMacOS::PlatformSDLMacOS()
-				: PlatformSDL("macos")
+		: PlatformSDL("macos")
 	{
-				std::error_code error_code;
-
-				uint32_t buffer_size = PATH_MAX;
-				_NSGetExecutablePath(m_ApplicationPath, &buffer_size);
-
-				m_LogicalDrivesList.push_back("/Volumes");
+		std::error_code error_code;
 
-				// Add the user's home folder
-				char *realHome = getpwuid(getuid())->pw_dir;
+		uint32_t buffer_size = PATH_MAX;
+		_NSGetExecutablePath(m_ApplicationPath, &buffer_size);
 
-				if(realHome != nullptr)
-				{
-						m_RealHome = std::string(realHome);
+		// Add the user's home folder
+		char* realHome = getpwuid(getuid())->pw_dir;
```

Comment:
Looks like the memory buffer isn't freed again. Can you verify how the memory of char* is handled. If it's not supposed to be user managed, is suggest making the pointer const.

Reply:
Ok. So checked the documentation. You do not own the memory returned from the call, and so it should be const and you're not supposed to free it.